tests/test_model.py

```diff
@@ -984,8 +1038,32 @@ def test_different_item_metadata_types_for_hybrid_pretrained_model(implicit_mode
     trainer_3 = CollieTrainer(model=model_3, logger=False, enable_checkpointing=False, max_epochs=1)
     trainer_3.fit(model_3)
 
+    # ensure that we end up with the same ``user_metadata`` regardless of the input type
+    model_4 = HybridPretrainedModel(train=train,
+                                    val=val,
+                                    user_metadata=user_metadata_df,
+                                    trained_model=implicit_model)
+    trainer_4 = CollieTrainer(model=model_4, logger=False, enable_checkpointing=False, max_epochs=1)
+    trainer_4.fit(model_1)
```

Comment:
Last change here, I promise: 

```suggestion
    trainer_4.fit(model_4)
```

Same for the test below this.